src/widget/systemtrayicon.cpp

```diff
@@ -116,6 +113,26 @@ QString SystemTrayIcon::extractIconToFile(QIcon icon, QString name)
     return iconPath;
 }
 
+#if defined(ENABLE_SYSTRAY_GTK_BACKEND) || defined(ENABLE_SYSTRAY_STATUSNOTIFIER_BACKEND)
+GdkPixbuf* SystemTrayIcon::convertQIconToPixbuf(const QIcon &icon)
+{
+    void (*callbackFreeImage)(guchar*, gpointer) =
+        [](guchar* image_bytes, gpointer)
+    {
+        delete[] image_bytes;
```

Comment:
hum… though, would that mean that any lambda use leads to creating a new instance?  that would make them a lot more costly than a normal function or even method :S
But yeah, I don't see how it could do otherwise if the user doesn't destroy the object, so if it works as a standard function pointer.